go-controller/pkg/util/libovsdb.go

```diff
@@ -0,0 +1,186 @@
+package util
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"time"
+
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+	"gopkg.in/fsnotify/fsnotify.v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+type ReconnectFunc func(client.Client) error
+
+// newClient creates a new client object given the provided config
+// the stopCh is required to ensure the two goroutines for ssl cert
+// update and reconnect are not leaked
+func newClient(cfg config.OvnAuthConfig, dbModel *model.DBModel, reconnectFn ReconnectFunc, stopCh <-chan struct{}) (client.Client, error) {
```

Comment:
A name like `onReconnectFn` would have helped me understand better what this is for.